test/Microsoft.Framework.PackageManager.FunctionalTests/DnuSourcesTests.cs

```diff
@@ -0,0 +1,154 @@
+        [MemberData(nameof(RuntimeComponents))]
+        public void DnuSources_NoSources(string flavor, string os, string architecture)
+        {
+            var environment = new Dictionary<string, string>
+            {
+                { "DNX_TRACE", "0" },
+            };
+
+            var rootConfig =
+@"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+  <packageSources>
+    <clear /> <!-- Remove the effects of any machine-level config -->
+  </packageSources>
+</configuration>";
+
+            var projectStructure =
+@"{
+    'root': {
+        'NuGet.Config': """",
+    }
+}";
+
+            var expectedContent = @"Registered Sources:
+1.  https://www.nuget.org/api/v2/ https://www.nuget.org/api/v2/
+";
+            var runtimeHomePath = _fixture.GetRuntimeHomeDir(flavor, os, architecture);
+            using (var testEnv = new DnuTestEnvironment(runtimeHomePath, projectName: "Project Name"))
+            {
+                var projectPath = testEnv.ProjectPath;
+                DirTree.CreateFromJson(projectStructure)
+                    .WithFileContents("root/NuGet.Config", rootConfig)
+                    .WriteTo(projectPath);
+
+                string output;
+                string error;
+                var exitCode = DnuTestUtils.ExecDnu(
+                    runtimeHomePath,
+                    subcommand: "list-sources",
+                    arguments: "root",
+                    stdOut: out output,
+                    stdErr: out error,
+                    environment: environment,
+                    workingDir: projectPath);
+
+                Assert.Equal(0, exitCode);
+                Assert.Empty(error);
+                Assert.Equal(expectedContent, output);
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(RuntimeComponents))]
+        public void DnuSources_ListsAllSources(string flavor, string os, string architecture)
+        {
+            var environment = new Dictionary<string, string>
+            {
+                { "DNX_TRACE", "0" },
+            };
+
+            var rootConfig =
+@"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+  <packageSources>
+    <clear /> <!-- Remove the effects of any machine-level config -->
+    <add key=""Source1"" value=""https://source1"" />
+    <add key=""Source2"" value=""https://source2"" />
+  </packageSources>
+</configuration>";
+
+            var subConfig =
+@"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+  <packageSources>
+    <add key=""Source3"" value=""https://source3"" />
+  </packageSources>
+  <disabledPackageSources>
+    <add key=""Source1"" value=""https://source1"" />
+  </disabledPackageSources>
+</configuration>";
+
+            var projectStructure =
+@"{
+    'root': {
+        'NuGet.Config': """",
+        'sub': {
+            'NuGet.Config': """"
+        }
+    }
+}";
+
+            var expectedContent = @"Registered Sources:
+1.  Source1 https://source1 [Disabled]
```

Comment:
Generally, yes. I do intentionally clear the machine level sources so that should make it deterministic. Changes to our package source code could adjust this order, but those should cause the tests to fail anyway.